crates/pathfinder/src/rpc/v02/method/add_deploy_account_transaction.rs

```diff
@@ -0,0 +1,132 @@
+use anyhow::Context;
+use serde_with::serde_as;
+
+use crate::rpc::serde::{FeeAsHexStr, TransactionVersionAsHexStr};
+use crate::{
+    core::{
+        CallParam, ClassHash, ContractAddress, ContractAddressSalt, Fee, StarknetTransactionHash,
+        TransactionNonce, TransactionSignatureElem, TransactionVersion,
+    },
+    rpc::v02::RpcContext,
+    sequencer::ClientApi,
+};
+
+const fn transaction_version_zero() -> TransactionVersion {
+    TransactionVersion(web3::types::H256::zero())
+}
+
+#[serde_as]
+#[derive(Clone, Debug, serde::Deserialize, serde::Serialize, PartialEq, Eq)]
+#[serde(deny_unknown_fields)]
+pub struct DeployAccountTransaction {
+    // Fields from BROADCASTED_TXN_COMMON_PROPERTIES
+    #[serde_as(as = "TransactionVersionAsHexStr")]
+    #[serde(default = "transaction_version_zero")]
+    pub version: TransactionVersion,
+    #[serde_as(as = "FeeAsHexStr")]
+    pub max_fee: Fee,
+    pub signature: Vec<TransactionSignatureElem>,
+    pub nonce: TransactionNonce,
+
+    // Fields from DEPLOY_ACCOUNT_TXN_PROPERTIES
+    pub contract_address_salt: ContractAddressSalt,
+    pub constructor_calldata: Vec<CallParam>,
+    pub class_hash: ClassHash,
+}
+
+#[derive(Clone, Debug, serde::Deserialize, serde::Serialize, PartialEq, Eq)]
+pub struct AddDeployAccountTransactionInput {
+    deploy_account_transaction: DeployAccountTransaction,
```

Comment:
This type has to be `BroadcastedTransaction` otherwise parsing the request will fail due to the `type` property being unknown.

Also, please add a test case that makes sure that a request conforming to the specification is parsed correctly. (See the `starknet_addDeclareTransaction` PR for an example.)

Reply:
@CHr15F0x had a better idea, so I've updated the declare txn PR.

Reply:
Got it. Updated in a way similar to `starknet_addDeclareTransaction`.